test/e2e/beta/metamask-beta-ui.spec.js

```diff
@@ -525,15 +521,6 @@ describe('MetaMask', function () {
       await delay(regularDelayMs)
     })
 
-    it('confirms a deploy contract transaction in the popup', async () => {
```

Comment:
Why is this now unnecessary?

Reply:
It was never necessary. It was merged with the network provider PR, I believe in an attempt to handle some of the stuck in submitted states we were seeing. In fact what this does is attempts to confirm an already confirmed transaction. (That such is possible may point to another issue regarding how quickly the state of an open popup gets updated... but it is a separate issue.)

Without the additions of 463-467 here https://github.com/MetaMask/metamask-extension/pull/5059/files#diff-b00929238ad0453812af55a40336e889R463, it was possible for the `confirms a deploy contract transaction` tests to pass without the confirmation happening, and then get stuck in a submitting state which would prevent the `calls and confirms a contract method where ETH is sent` to fail. I believe these `confirms a deploy contract transaction in the popup` tests were an attempt to account for that.

Such an attempt is not a bad guess because the popup is open at this point in the test run. It is only closed further below in `calls and confirms a contract method where ETH is received` with `await closeAllWindowHandlesExcept(driver, [extension, dapp])`. For now, however, we want to continue the strategy of avoiding interaction with the popup as much as possible because of firefox woes.

As it stands, including the `confirms a deploy contract transaction in the popup` tests can cause the tests to fail and was a contributer to the uptick in false negatives we saw towards the end of last week.

Reply:
Thanks for the detailed response! 💯